# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/fetch/Fetch.cpp
# Commit: a8c5441fd896
# Full Hash: a8c5441fd89658dea45d939b1e930cf3d148fafd
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2024-01-26 21:47:24
# Description:
#   Bug 1873573 - Rename PromiseWorkerProxy::WorkerPromise() to GetWorkerPromise(). r=dom-worker-reviewers,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D199323
# ==============================================================================

diff -r 533ecc686bac -r a8c5441fd896 dom/fetch/Fetch.cpp
--- a/dom/fetch/Fetch.cpp	Fri Jan 26 10:41:43 2024 +0000
+++ b/dom/fetch/Fetch.cpp	Fri Jan 26 10:48:56 2024 +0000
@@ -279,7 +279,7 @@
     aWorkerPrivate->AssertIsOnWorkerThread();
     MOZ_ASSERT(!mIsShutdown);
 
-    return mPromiseProxy->WorkerPromise();
+    return mPromiseProxy->GetWorkerPromise();
   }
 
   FetchObserver* GetFetchObserver(WorkerPrivate* aWorkerPrivate) const {
@@ -806,7 +806,10 @@
     }
 
     RefPtr<Promise> promise = mResolver->WorkerPromise(aWorkerPrivate);
-    MOZ_ASSERT(promise);
+    // Once Worker had already started shutdown, workerPromise would be nullptr
+    if (!promise) {
+      return true;
+    }
     RefPtr<FetchObserver> fetchObserver =
         mResolver->GetFetchObserver(aWorkerPrivate);
 